libflux/src/semantic/mod.rs

```diff
@@ -37,6 +49,14 @@ pub fn analyze_source(source: &str) -> Result<nodes::Package> {
         package: "main".to_string(),
         files: vec![file],
     };
-    analyze(ast_pkg)
-    // TODO infer
+    let mut sem_pkg = analyze(ast_pkg)?;
+    // TODO(affo): add a stdlib Importer.
+    let (_, sub) = infer_pkg_types(
+        &mut sem_pkg,
+        Environment::empty(),
+        &mut Fresher::new(),
+        &Importer::new(),
+    )?;
```

Comment:
I thought `analyze` took a `Fresher` argument? I think this is the reason the below test case isn't passing. `analyze` and `infer_pkg_types` must take the same mutable `Fresher` reference.